### ISSUE TITLE
Support for "ultra strong" 6581 saw+pulse?

### ISSUE THREAD

**Issue by acrouzet** [libsidplayfp/libsidplayfp#140]
Some 6581s retain the MSB when saw+pulse is selected, resulting in a much lower and louder waveform, as heard here: https://youtu.be/7slZAeyS6dQ?si=GUwIDID8KkxitpAw

Since adjustable combined waves were recently implemented, I'm wondering if this could/should be taken into account in the adjustments.

**Comment by drfiemost** [comment]
Currently this is done unconditionally for the 6581:

https://github.com/libsidplayfp/libsidplayfp/blob/9cc3421737837558d360105dcc9ace9eb7c500f9/src/builders/residfp-builder/residfp/WaveformGenerator.h#L401-L406

Is this something chip dependent, meaning that it happening consistently on certain chips?
In that case it can be made configurable.

**Comment by acrouzet** [comment]
This is where it gets a bit odd. I just came across a simple BASIC program that sweeps through each wave sequentially.
The webpage it's on features some recordings: http://ploguechipsounds.blogspot.com/2010/05/one-page-basic-sid-benchmark.html

As you can hear, wave 6 exhibits a loud crackling sound on some of the 6581s. I also had my friend record the program on his 6581, [and it sounds even worse](https://drive.google.com/file/d/1vOoHKZsJLydfzNKXYzqYOdRkoLsLA2a5/view?usp=drive_link)

Looking at the waveforms of the recordings closely, it really does seem like the distortion is the MSB switching on and off rapidly.

Perhaps this is something to do with the specifics of how the BASIC program is writing to the SID, since I haven't heard such saw+pulse distortion in normal tunes on any 6581s. Still, this must mean something.

**Comment by reFX-Mike** [comment]
I don't think this is desirable to emulate. No SID tune composer would try to take advantage of something that sounds horrible and at the same time only affects a tiny portion of their listeners.

It's still undesirable even if we ignore the context of SID tunes. It may be a scientific curiosity, but it's neither pleasant nor useful.

Just because we can, doesn't mean we should.

**Comment by acrouzet** [comment]
Basically, to me this means I don't know if this MSB enabling is even consistent on a single chip anymore. More research is needed.

**Comment by drfiemost** [comment]
It says only P+S, and not every voice.
Quoting a comment:

> Anonymous[ May 31, 2020 at 8:02 AM](https://ploguechipsounds.blogspot.com/2010/05/one-page-basic-sid-benchmark.html?showComment=1590937374179#c1290856719993245777)
The _PS_ test seems to trip up the most chips by far - I've heard loads of crackling from a couple of mine (R3 and R4AR) and a couple have this odd digital 'ping' noise at the end exactly as in your R3 example. Should this be considered a fault? I've got two R3s here which both do it, one on voice 3, the other on voices 1 & 3. I have several R4ARs and an R2 that don't do it at all.
